assembly/src/ast/procedure/name.rs

```diff
@@ -311,10 +333,12 @@ impl FromStr for ProcedureName {
                     break Err(IdentError::InvalidChars { ident: s.into() });
                 }
             },
-            Some((_, c)) if c.is_ascii_lowercase() || c == '_' || c == '$' => {
+            Some((_, c))
+                if c.is_ascii_lowercase() || c == '_' || c == '-' || c == '$' || c == '.' =>
+            {
```

Comment:
Not for this PR, but the code in this function is a bit difficult to follow. Let's create an issue (could be a "good first issue") to make the code more readable.

Reply:
I agree. I made https://github.com/0xPolygonMiden/miden-vm/issues/1663 for it.